internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -160,6 +162,48 @@ func (s *svc) ListReceivedShares(ctx context.Context, req *collaboration.ListRec
 	if err != nil {
 		return nil, errors.Wrap(err, "gateway: error calling ListReceivedShares")
 	}
+
+	// TODO: This is a hack for now.
+	// Can we do that cleaner somehow?
```

Comment:
Potentially we could use events to propagate e-tags also to share roots. But I'm not entirely sure that's feasible.

Reply:
That'd be possible, but required some CS3 API changes to somehow include the etag in a share.

Reply:
Yes. I was only about possible future improvements. The hack is good for now :+1: